rpc/interface/src/route/json_rpc.rs

```diff
@@ -0,0 +1,68 @@
+//! JSON-RPC 2.0 endpoint route functions.
+
+//---------------------------------------------------------------------------------------------------- Import
+use std::borrow::Cow;
+
+use axum::{extract::State, http::StatusCode, Json};
+use tower::ServiceExt;
+
+use cuprate_json_rpc::{
+    error::{ErrorCode, ErrorObject},
+    Id,
+};
+use cuprate_rpc_types::{
+    json::{JsonRpcRequest, JsonRpcResponse},
+    RpcCallValue,
+};
+
+use crate::{rpc_handler::RpcHandler, rpc_request::RpcRequest, rpc_response::RpcResponse};
+
+//---------------------------------------------------------------------------------------------------- Routes
+/// The `/json_rpc` route function used in [`crate::RouterBuilder`].
+pub(crate) async fn json_rpc<H: RpcHandler>(
+    State(handler): State<H>,
+    Json(request): Json<cuprate_json_rpc::Request<JsonRpcRequest>>,
+) -> Result<Json<cuprate_json_rpc::Response<JsonRpcResponse>>, StatusCode> {
+    // TODO: <https://www.jsonrpc.org/specification#notification>
+    //
+    // JSON-RPC notifications (requests without `id`)
+    // must not be responded too, although, the request's side-effects
+    // must remain. How to do this considering this function will
+    // always return and cause `axum` to respond?
```

Comment:
This is (maybe) one of the differences that will be documented in the architecture book: https://www.jsonrpc.org/specification#notification.

On one hand, `monerod` ignores this and following that behavior is easier, on the other hand, the specification very clearly states this is incorrect:

> The Server MUST NOT reply to a Notification

My first thought is to test bitcoin/eth/zcash and see if they follow the spec, if so, I think we should as well.

Reply:
Yeah I think I agree, the spec does seem pretty firm there.

However it is probably not worth the effort if monerod is already breaking spec.